### PR TITLE
List method for QuizzesController

### PR DESCRIPTION
**Pull request by mironovandrei**
- [x] list method

- [x] test list method

- [x] ACL
#362

### CODE REVIEW

**Comment by mironovandrei** [comment]
@malinink PR ready for review

File: src/test/java/easytests/api/v1/controllers/QuizzesControllerTest.java [hunk 1]
```diff
@@ -1,12 +1,144 @@
...
+            final QuizModel quizModel = new QuizModel();
+            quizModel.map(this.quizzesSupport.getEntityFixtureMock(idx));
+
+            TesteeModelInterface testeeModel = new TesteeModel();
+            testeeModel.map(testeesSupport.getEntityFixtureMock(0));
+            testeeModel.setQuiz(quizModel);
+            quizModel.setTestee(testeeModel);
+
+            if (quizModel.getId().equals(1)) {
+                quizzesModels.add(quizModel);
```
Comment (malinink) [inline src/test/java/easytests/api/v1/controllers/QuizzesControllerTest.java]: seems that you trying to create list, but than you filter it to 1 element.
make list consist of two items, one with testee included other without it

File: src/main/java/easytests/api/v1/mappers/QuizzesMapper.java [hunk 2]
```diff
@@ -1,12 +1,34 @@
...
+    public QuizzesMapper() {
+        super();
+        this.createTypeMap(QuizModel.class, Quiz.class)
+                .addMappings(mapper -> mapper.when(context -> !(context.getSource() instanceof ModelsListEmpty))
+                        .<Integer>map(quizModel -> quizModel.getTestee().getQuiz().getId(),
+                            (quiz, id) -> quiz.getTestee().getQuiz().setId(id)))
+                .addMappings(mapper -> mapper.when(context -> !(context.getSource() instanceof ModelsListEmpty))
+                        .<Integer>map(quizModel -> quizModel.getTestee().getId(),
+                            (quiz, id) -> quiz.getTestee().setId(id)))
+                .addMappings(mapper -> mapper.using(src -> ((LocalDateTime) src.getSource())
```
Comment (malinink) [inline src/main/java/easytests/api/v1/mappers/QuizzesMapper.java]: how that handle nulls?

File: src/main/java/easytests/api/v1/mappers/QuizzesMapper.java [hunk 3]
```diff
@@ -1,12 +1,34 @@
...
+                        .format(DateTimeFormatter.ISO_LOCAL_DATE_TIME) + Z)
+                        .<String>map(QuizModel::getStartedAt, (quiz, startedAt) -> quiz.setStartedAt(startedAt)))
+                .addMappings(mapper -> mapper.using(src -> ((LocalDateTime) src.getSource())
+                        .format(DateTimeFormatter.ISO_LOCAL_DATE_TIME) + Z)
```
Comment (malinink) [inline src/main/java/easytests/api/v1/mappers/QuizzesMapper.java]: also that

File: src/test/java/easytests/api/v1/controllers/QuizzesControllerTest.java (hunk 1 above)
Comment (malinink) [reply src/test/java/easytests/api/v1/controllers/QuizzesControllerTest.java]: Add nulls values (where possible) into test case.

**Review by malinink — CHANGES_REQUESTED** [review]

**Comment by mironovandrei** [comment]
@malinink ready for review

**Comment by malinink** [comment]
All is OK, but! It seems to me that we forget about Issue model that is linked to Quiz.

@miron97 
Have you time to add it as required field into quiz and add it into model, list method and fix tests?

**Comment by mironovandrei** [comment]
@malinink 
Ok, I think it`s not difficult

**Comment by mironovandrei** [comment]
@malinink ready

**Comment by malinink** [comment]
@miron97 Thanks!

**Review by malinink — APPROVED** [review]

**Pull request merged**